WebDriverAgentLib/Categories/XCUIElement+FBTyping.m

```diff
@@ -141,24 +142,33 @@ - (BOOL)fb_clearTextWithSnapshot:(FBXCElementSnapshotWrapper *)snapshot
   NSString *placeholderValue = snapshot.placeholderValue;
   NSUInteger preClearTextLength = [currentValue fb_visualLength];
   do {
-    if (retry >= MAX_CLEAR_RETRIES - 1) {
-      // Last chance retry. Tripple-tap the field to select its content
-
-      if ([self respondsToSelector:@selector(tapWithNumberOfTaps:numberOfTouches:)]) {
-        // e.g. tvOS 17 raised unrecognized selector error for XCUIElementTypeSearchField
-        // while following typeText worked.
-        [self tapWithNumberOfTaps:3 numberOfTouches:1];
-      }
-      return [FBKeyboard typeText:backspaceDeleteSequence error:error];
-    }
-
-    NSString *textToType = [backspaceDeleteSequence fb_repeatTimes:preClearTextLength];
     if (shouldPrepareForInput && 0 == retry) {
       [self fb_prepareForTextInputWithSnapshot:snapshot];
     }
-    if (![FBKeyboard typeText:textToType error:error]) {
+    NSString *backspacesToType = [backspaceDeleteSequence fb_repeatTimes:preClearTextLength];
+#if TARGET_OS_IOS
+    if (retry == MAX_CLEAR_RETRIES - 2) {
+      // https://github.com/appium/appium/issues/19389
+      [[XCUIDevice sharedDevice] fb_performIOHIDEventWithPage:0x07
```

Comment:
This method works only when the field has a focus like by tap. So... what do you think about adding a tap once then `fb_performIOHIDEventWithPage` as 2nd retrial, or adding tap before `fb_performIOHIDEventWithPage`.

So,

1 st attempt -> fb_performIOHIDEventWithPage only
2nd -> tap the element once, then fb_performIOHIDEventWithPage
3rd -> `tapWithNumberOfTaps:3`
4th -> typeText

or 

1st -> tap the element once, then fb_performIOHIDEventWithPage
2nd -> `tapWithNumberOfTaps:3`
3rd -> typeText